### PR TITLE
Add lint-proto job to CI workflow

### PR DESCRIPTION
**Pull request by devholic**
Forgot to add linting proto files in CI workflow 😓

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,6 +16,7 @@ CERT_MANAGER_CHART_VERSION := 1.11.0
 
 .PHONY: lint
 lint:
+	go mod vendor
```
Comment (devholic) [inline Makefile:19]: We have to make sure `gogoproto` exists in vendor directory, since we use `gogoproto`.

**Comment by krancour** [comment]
`make lint` is surfacing a lot of things that CI isn't catching. Things like:

```
vendor/k8s.io/kube-aggregator/pkg/apis/apiregistration/v1beta1/generated.proto:123:18:Field name "versionPriority" should be lower_snake_case, such as "version_priority".
```

I think this is because CI is only linting the `api/` dir and the `buf lint` command in the `lint` make target seems to be linting everything.

File: Makefile (hunk 1 above)
Comment (devholic) [reply Makefile]: I split into `lint-go` and `lint-proto` 🙂

**Review by krancour — APPROVED** [review]
LGTM

**Pull request merged**